### PR TITLE
[filesystem] ZipManager: skip path traversal

### PR DESCRIPTION
**Pull request by Rechi**
## Description
Skip items in a zip file, which try to traverse to a parent directory.

## Motivation and Context
Without this a zip file can override every file the current user has write permission.

## How Has This Been Tested?
Tested with a malicious zip file.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

@ace20022 @MartijnKaijser @wsnipex FYI

### CODE REVIEW

**Comment by wsnipex** [comment]
thanks! jenkins build this please

**Comment by tamland** [comment]
Please add some tests that verifies this.

**Comment by Rechi** [comment]
@tamland should I add tests which only check the regex or also check with a real zip file?

**Comment by tamland** [comment]
I think just the regex/file filtering should be good enough.
Not sure how smart it would be to unpack a real malicious zip to a real directory...

**Comment by Rechi** [comment]
added some testcases
Still not sure if I create a new testcase file is valid.

File: xbmc/filesystem/ZipManager.h [hunk 1]
```diff
@@ -32,12 +32,15 @@
...
 
 #include <memory.h>
+#include <regex>
 #include <string>
 #include <vector>
 #include <map>
 
 class CURL;
 
+static const std::regex PATH_TRAVERASL("(^|\\/|\\\\)\\.{2}($|\\/|\\\\)");
```
Comment (Paxxi) [inline xbmc/filesystem/ZipManager.h]: spelling error 😄 
Also you can use a raw string `R"_((^|\/|\\)\.{2}($|\/|\\))_"` a bit messy but makes the regex a bit simpler to read

**Comment by Rechi** [comment]
@Paxxi updated

**Review by Paxxi — COMMENTED** [review]
Looks good to me

**Comment by Rechi** [comment]
@tamland @wsnipex please have a look again as I've done a few changes

**Comment by wsnipex** [comment]
looks good

**Comment by Rechi** [comment]
jenkins build this please

**Review by tamland — APPROVED** [review]

**Pull request merged**